### DISCUSSION TITLE
Upgrading?

### DISCUSSION THREAD

**Discussion by kanflo** [greghesp/ha-bambulab#638] [General]
I feel a bit silly for asking, but how do I upgrade the integration? HASS -> HACS tells me version 2.0.21 is the "version downloaded". Clicking "Open HACS Repository on my ..." asks me if I want to "Add custom repository" but after clicking "yes" I am still at 2.0.21.

**Comment by AdrianGarside** [comment]
HACS 2.0 changed the experience quite a bit so I'm only describing that version.

Common case is that HACS has auto-detected a newer version is available and will list it in the top 'Pending updates' section. I'm up fully up to date so currently is shows in the next section for me when I click the HACS button on the left pane in home assistant:
![image](https://github.com/user-attachments/assets/554432ad-6185-4700-b126-ba447f4ee717)

Click the Bambu Labs entry, then the ... button top right, then the redownload option:
![image](https://github.com/user-attachments/assets/effb4989-3e2e-4f29-b4d7-a68ead6e5757)

This should then default to either the latest release or latest pre-release version depending if you've configured HACS to show you prerelease version for this integration. There's also a drop down showing all versions if you need to downgrade for any reason:
![image](https://github.com/user-attachments/assets/f808354e-483a-4717-b2ae-168a0f2c3a3f)

After it downloads (can take a while), you do need to restart home assistant. HACS used to make the really obvious but now it's just a notification that shows in home assistant settings and it can be easy to miss.

**Reply by kanflo** [reply]
Thanks! I seem to be running a rather old version of HACS so I will start with upgrading that.
